In [28]:
import numpy as np
import scipy.stats as st

# Problema 1
Se hizo una encuesta a 325 personas mayores de 16 años y se encontró que 120 iban al teatro regularmente. Hallar, con un nivel de confianza del 94%, un intervalo para estudiar la proporción de los ciudadanos que van al teatro regularmente.

In [29]:
# Datos
n = 325
x = 120
confianza = 0.94

In [30]:
proporcionMuestral = x/n
alfa = 1 - confianza
z = st.norm.ppf(1 - alfa/2)

print("Proporción muestral:", proporcionMuestral)
print("Alfa:", alfa)
print("Valor crítico z:", z)

Proporción muestral: 0.36923076923076925
Alfa: 0.06000000000000005
Valor crítico z: 1.8807936081512509


In [31]:
errorEstandar = np.sqrt(proporcionMuestral * (1 - proporcionMuestral) / n)

print("Error estándar:", errorEstandar)

Error estándar: 0.026769638846104032


In [32]:
limiteInferior = proporcionMuestral - z * errorEstandar
limiteSuperior = proporcionMuestral + z * errorEstandar

print("Intervalo de confianza al 94%: [{:.4f}, {:.4f}]".format(limiteInferior, limiteSuperior))

Intervalo de confianza al 94%: [0.3189, 0.4196]


Con un 94% de confianza, la proporción de ciudadanos que van al teatro regularmente está entre 32% y 41.6%

# Problema 2
En una encuesta realizada a 800 personas elegidas al azar del censo electoral, 240 declaran su intención de votar por el partido A.
1. Estimar, con un nivel de confianza del 95,45% entre qué valores se encuentra la intención de voto a dicho partiddo en todo el censo.
2. Discutir, razonablemnte, el efecto que tendría sobre el intervalo de confianza el aumento, o la disminución, del nivel de confianza.

In [33]:
# Datos
n = 800
x = 240
confianza = 0.9545

In [34]:
proporcionMuestral = x/n
alfa = 1 - confianza
z = st.norm.ppf(1 - alfa/2)

print("Proporción muestral:", proporcionMuestral)
print("Alfa:", alfa)
print("Valor crítico z:", z)

Proporción muestral: 0.3
Alfa: 0.045499999999999985
Valor crítico z: 2.0000024438996027


In [35]:
errorEstandar = np.sqrt(proporcionMuestral * (1 - proporcionMuestral) / n)

print("Error estándar:", errorEstandar)

Error estándar: 0.01620185174601965


In [36]:
# Cálculo del intervalo de confianza
limiteInferior = proporcionMuestral - z * errorEstandar
limiteSuperior = proporcionMuestral + z * errorEstandar

print("Intervalo de confianza al 95.45%: [{:.4f}, {:.4f}]".format(limiteInferior, limiteSuperior))

Intervalo de confianza al 95.45%: [0.2676, 0.3324]


Con un 95,45% de confianza, la proporción de intención de voto por el partido A está entre 26.8% y 33,2%

# Problema 3
En un control de calidad se analizó una muestra aleatoria de 750 tornillos resultando defectuosos 80 de ellos.
1. Hallar un intervalo de confianza para la proporción de tornillos defectuosos en el conjunto de producción con 99% de confianza.
2. ¿Cuál es el error máximo contenido en la estimación anterior?
3. Si deseamos que el error cometido, con el mismo nivel de confianza, sea la décima parte del apartado anterior, ¿cuál ha de ser el tamaño de la muestra?

In [37]:
# Datos
n = 750
x = 80
confianza = 0.99

In [38]:
proporcionMuestral = x/n
alfa = 1 - confianza
z = st.norm.ppf(1 - alfa/2)

print("Proporción muestral:", proporcionMuestral)
print("Alfa:", alfa)
print("Valor crítico z:", z)

Proporción muestral: 0.10666666666666667
Alfa: 0.010000000000000009
Valor crítico z: 2.5758293035489004


In [39]:
errorEstandar = np.sqrt(proporcionMuestral * (1 - proporcionMuestral) / n)
print("Error estándar:", errorEstandar)

Error estándar: 0.01127172798872701


In [40]:
limiteInferior = proporcionMuestral - z * errorEstandar
limiteSuperior = proporcionMuestral + z * errorEstandar

print("Intervalo de confianza al 99%: [{:.4f}, {:.4f}]".format(limiteInferior, limiteSuperior))

Intervalo de confianza al 99%: [0.0776, 0.1357]


In [41]:
# Error máximo
errorMaximo = z * errorEstandar
print("Error máximo permitido:", errorMaximo)

Error máximo permitido: 0.029034047254995345


In [42]:
errorNuevo = errorMaximo / 10
muestraRequerida = (z**2 * proporcionMuestral * (1 - proporcionMuestral) / errorNuevo**2)

print("Tamaño de muestra requerido para un error de {:.4f}: {:.0f}".format(errorNuevo, muestraRequerida))

Tamaño de muestra requerido para un error de 0.0029: 75000


1. Con un 99% de confianza, la proporción de tornillos defectuosos está entre 8% y 13,3%
2. El error máximo es 2,67%
3. La muestra necesaria para reducir el error es de 75000